### PR TITLE
register cloudstack datacenter reconciler

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*
Build and register new datacenter reconciler for CloudStack

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: controllers/cloudstack_datacenter_controller.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func (r *CloudStackDatacenterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&anywherev1.CloudStackDatacenterConfig{}).
+		Complete(r)
+}
+
+// Reconcile implements the reconcile.Reconciler interface.
+func (r *CloudStackDatacenterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	// TODO fetch CloudStack datacenter object and implement reconcile
+	return ctrl.Result{}, nil
```
Comment (cxbrowne1207) [inline controllers/cloudstack_datacenter_controller.go:34]: Should this be implemented for this task or is there another task on the board for it?

Comment (cxbrowne1207) [inline controllers/cloudstack_datacenter_controller.go:34]: Can we add a test for this that would be updated by whoever implements this?

Comment (tatlat) [reply controllers/cloudstack_datacenter_controller.go:34]: Yes, there should be one.

**Pull request merged**